src/Mvc/Mvc.Core/src/Formatters/SystemTextJsonInputFormatter.cs

```diff
@@ -98,5 +131,68 @@ private Stream GetInputStream(HttpContext httpContext, Encoding encoding)
 
             return new TranscodingReadStream(httpContext.Request.Body, encoding);
         }
+
+        // Keep in sync with NewtonsoftJsonInputFormatter.GetPathMetadata
+        private ModelMetadata GetPathMetadata(ModelMetadata metadata, string path)
+        {
+            var index = 0;
+            while (index >= 0 && index < path.Length)
+            {
+                if (path[index] == '[')
```

Comment:
What about escaped strings in a path? e.g.

`$.DictionaryContainer.['it\'s a lot of work to handle all situations'].Name`

This can happen when the path has a dictionary key in it and the dictionary key requires escaping.

Reply:
Logic in serializer: https://github.com/dotnet/corefx/blob/94c5f1384c9b6477d164050de964cf6dcf99b2a9/src/System.Text.Json/src/System/Text/Json/Serialization/ReadStack.cs#L105-L123

Reply:
Hmm, I think you're going to need to duplicate logic for reading an escaped JSON string. Maybe you could reuse Utf8JsonReader but you'd need to parse the string anyway to find where it is closed so you had a slice of text to pass to it.

Reply:
I'm just here so I won't get fined... and to say that the existing behaviour (trying to map JSON path to specific .NET properties) isn't useful for anyone.

Reply:
Ok. I'm ok with us starting off without the mapping behavior for System.Text.Json. It's different from Json.NET, and we can consider putting that behind a flag.

Removing this would mean there's less of this weird code around and the ModelState \ `ValidationProblemDetails` that's returned has JSONPath in it. That sound reasonable?